v-next/hardhat-keystore/src/internal/hook-handlers/configuration-variables.ts

```diff
@@ -0,0 +1,50 @@
+import type { KeystoreLoader } from "../types.js";
+import type { ConfigurationVariable } from "@ignored/hardhat-vnext/types/config";
+import type {
+  ConfigurationVariableHooks,
+  HookContext,
+} from "@ignored/hardhat-vnext/types/hooks";
+
+import { FileManagerImpl } from "../loaders/file-manager.js";
+import { KeystoreFileLoader } from "../loaders/keystore-file-loader.js";
+
+export default async (): Promise<Partial<ConfigurationVariableHooks>> => {
+  // Use a cache with hooks since they may be called multiple times consecutively.
+  let keystoreLoader: KeystoreLoader | undefined;
+
+  const handlers: Partial<ConfigurationVariableHooks> = {
+    fetchValue: async (
+      context: HookContext,
+      variable: ConfigurationVariable,
+      next,
+    ) => {
+      if (keystoreLoader === undefined) {
+        keystoreLoader =
+          await _setupLoaderWithContextBasedUserInterruptions(context);
+      }
+
+      if (!(await keystoreLoader.isKeystoreUninitialized())) {
+        return next(context, variable);
+      }
+
+      const keystore = await keystoreLoader.loadKeystore();
```

Comment:
Do we want to guard against file loading errors (i.e. permissions problems) here and return the next value, or do we expose those exceptions immediately?

Currently we fail early.